Date inputs can be configured to show the year

HarryAmmon opened the pull request:
Adds a tag helper and Umbraco configuration to hide the 'year' input box as part of a date input. This extends the work already done to hide the 'day' field. 

Includes some changes to the model binder, as there were a few bugs for when a year was not provided.

Defaults the year to 1900 after discussions with the team. I could provide a way to configure the default year, but that could come as an enhancement of this work.

Commit:
Create a tag helper to show the year, and hide the year if this is false AB#250195

Commit:
Add show years as a property in Umbraco for the date input block, update model binding to allow for dates without a provided year AB#250195

Commit:
Add an example of a date input without a year AB#250195

Commit:
Fix failing unit test and add unit test for day month only scenario AB#250195

sussexrick committed:
Merge branch 'develop' into feature/the-forgotten-year

sussexrick commented on GovUk.Frontend.Umbraco.ExampleApp/uSync/v9/ContentTypes/govukdateinputsettings.config:
```diff
@@ -48,6 +48,22 @@
+      <Type>Umbraco.TrueFalse</Type>
+      <Mandatory>false</Mandatory>
+      <Validation></Validation>
+      <Description><![CDATA[If the year field is hidden the defaults to 1900.]]></Description>
```
"the year defaults to"?

sussexrick commented on GovUk.Frontend.Umbraco/ModelBinding/UmbracoDateInputModelBinder.cs:
```diff
@@ -241,8 +255,15 @@ internal static DateInputParseErrors Parse(DateInputItemTypes itemTypes, string?
+                        assumedYear = parsedYear;
+                    }
+                    /*var assumedYear = yearIsValid ? parsedYear : 2000; */ 
+                    // Bug is here, the year could be valid because it has not been provided, which means parsedYear will be null
```
Looks like some unfinished work here?

sussexrick reviewed (commented):
Seems to work well and just minor comments added. However this is an important change and the ASP.NET example app should have an example too.

Commit:
Remove code comments for debugging AB#250195

Commit:
Fix typo AB#250195

Commit:
Merge branch 'feature/the-forgotten-year' of https://github.com/thepensionsregulator/govuk-frontend-aspnetcore-extensions into feature/the-forgotten-year

HarryAmmon commented:
> Seems to work well and just minor comments added. However this is an important change and the ASP.NET example app should have an example too.

I have addressed your comments. 

I'm unable to add an example to the ASP.NET example app as the changes are in the `GovUk.Frontend.Umbraco` namespace, which that project doesn't have access to. The ASP.NET example app does not contain an example for 'show-day', and justification for not doing that was included in the original PR #249.

sussexrick commented:
Things have changed since then, as the base [`DateInputModelBinder` supports both](https://github.com/x-govuk/govuk-frontend-aspnetcore/blob/main/src/GovUk.Frontend.AspNetCore/ModelBinding/DateInputModelBinder.cs) turning off the day field that we already supported and turning off the year field that you're implementing now.

That said, you're following our current approach well so I'm happy to approve and add aligning with the base project as a tech debt issue for the future.
